src/prep_data.R

```diff
@@ -1,20 +1,21 @@
-add_flow_condition <- function(data_in, date_start, date_end){
+
+#' @description Bin percentile data into flow condition categories
+#' @param data_in 1 month of streamflow percentiles generated from `gage-conditions-gif` pipeline
+#' @param date_start first day of focal month
+#' @param date_end last day of focal month
+#' @param breaks Percentile values to bin data at
+add_flow_condition <- function(data_in, date_start, date_end, breaks){
   data_in %>% 
-    mutate(date = as.Date(dateTime),
-           cond = 
-             case_when(
-               per < 0.05 ~ 'Driest',
-               per >= 0.05 & per < 0.1 ~ "Drier",
-               per >= 0.1 & per < 0.25 ~ "Dry", 
-               per >= 0.25 & per < 0.75 ~ "Normal",
-               per >= 0.75 & per < 0.9 ~ "Wet", 
-               per >= 0.9 & per < 0.95 ~ "Wetter", 
-               per >= 0.95 ~ "Wettest"
-             ),
-           cond = factor(cond, ordered = TRUE, levels = c("Driest", "Drier", "Dry", "Normal","Wet","Wetter", "Wettest"))) %>%
-    filter(date >= date_start, date <= date_end, !is.na(cond)) 
+    mutate(date = as.Date(dateTime)) %>%
+    filter(date >= date_start, date <= date_end, !is.na(per)) %>%
+    mutate(percentile_bin = cut(per, breaks = breaks, include.lowest = TRUE),
+           percentile_cond = factor(percentile_bin, labels = c("Driest", "Drier", "Dry", "Normal","Wet","Wetter", "Wettest")))
```

Comment:
All that being said, nice application here. Putting this in my pocket for the future